js/private/js_info.bzl

```diff
@@ -3,48 +3,71 @@
-        "transitive_sources": "A list of source files produced by the target and the target's transitive deps",
+        "declarations": "A depset of declaration files produced by the target",
+        "npm_linked_package_files": "A depset of files that are that are in npm linked package dependencies of this target",
+        "npm_linked_packages": "A depset of NpmLinkedPackageInfo providers that are dependencies of this target",
+        "npm_package_store_deps": "A depset of NpmPackageStoreInfo providers from npm dependencies of this target and the target's transitive dependencies to use as direct dependencies when linking downstream npm_package targets with npm_link_package",
+        "sources": "A depset of source files produced by the target",
+        "transitive_declarations": "A depset of declaration files produced by the target and the target's transitive deps",
+        "transitive_npm_linked_package_files": "A depset of files that are in npm linked package dependencies of this target and the target's transitive deps",
+        "transitive_npm_linked_packages": "A depset of NpmLinkedPackageInfo providers that are dependencies of this target and the target's transitive deps",
+        "transitive_sources": "A depset of source files produced by the target and the target's transitive deps",
     },
 )
 
 def js_info(
-        declarations = [],
-        npm_linked_packages = [],
-        npm_package_stores = [],
-        sources = [],
-        transitive_declarations = [],
-        transitive_npm_linked_packages = [],
-        transitive_npm_package_stores = [],
-        transitive_sources = []):
+        declarations = None,
+        npm_linked_package_files = None,
+        npm_linked_packages = None,
+        npm_package_store_deps = None,
+        sources = None,
+        transitive_declarations = None,
+        transitive_npm_linked_package_files = None,
+        transitive_npm_linked_packages = None,
+        transitive_sources = None):
     """Construct a JsInfo
 
     Args:
         declarations: See JsInfo documentation
+        npm_linked_package_files: See JsInfo documentation
         npm_linked_packages: See JsInfo documentation
-        npm_package_stores: See JsInfo documentation
+        npm_package_store_deps: See JsInfo documentation
         sources: See JsInfo documentation
         transitive_declarations: See JsInfo documentation
+        transitive_npm_linked_package_files: See JsInfo documentation
         transitive_npm_linked_packages: See JsInfo documentation
-        transitive_npm_package_stores: See JsInfo documentation
         transitive_sources: See JsInfo documentation
 
     Returns:
         A JsInfo provider
     """
+    if type(declarations) == "list":
```

Comment:
Do you think we'll always have this or drop it in 2.0? If drop then lets add a comment?

Reply:
its useful for downstream rules if you just have a list of files you can pass them in without wrapping them in a depset; otoh, downstream rules could write `depset(files).to_list()` and it would be a perf hit that would be hard to catch in the code